modules/3d-tiles/src/parsers/helpers/normalize-3d-tile-positions.js

```diff
@@ -1,24 +1,47 @@
 import {Vector3} from 'math.gl';
+import {GL} from '@loaders.gl/math';
 
-const scratchPosition = new Vector3();
+// Prepare attribute for positions
+// For quantized posititions, either expand to Float32Array or return custom accessor
+// https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/specification/TileFormats/Instanced3DModel/README.md#quantized-positions
 
-export function normalize3DTilePositionAttribute(tile, positions) {
-  if (tile.isQuantized) {
-    // https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/specification/TileFormats/Instanced3DModel/README.md#quantized-positions
+export function normalize3DTilePositionAttribute(tile, positions, options) {
+  if (!tile.isQuantized) {
+    return positions;
+  }
+
+  // Optionally decode quantized positions on GPU, for renderers that can't work with normalized attributes
+  if (options.decodeQuantizedPositions) {
+    tile.isQuantized = false;
+    return decodeQuantizedPositions(tile, positions);
+  }
+
+  // Use normalized shorts directly, no copying/processing.
+  // NOTE: The "missing" offset/scaling operations are automatically added to modelMatrix if `tile.isQuantized === true`
+  return {
+    type: GL.UNSIGNED_SHORT,
+    value: positions,
+    size: 3,
+    normalized: true
+  };
+}
+
+// Pre-scale quantized positions on CPU
+function decodeQuantizedPositions(tile, positions) {
+  const scratchPosition = new Vector3();
+  const decodedArray = new Float32Array(tile.pointCount * 3);
+
+  for (let i = 0; i < tile.pointCount; i++) {
     // POSITION_QUANTIZED * QUANTIZED_VOLUME_SCALE / 65535.0 + QUANTIZED_VOLUME_OFFSET
-    const decodedArray = new Float32Array(tile.pointCount * 3);
-    for (let i = 0; i < tile.pointCount; i++) {
-      scratchPosition
-        .set(positions[i * 3], positions[i * 3 + 1], positions[i * 3 + 2])
-        .multiply(tile.quantizedVolumeScale)
-        .scale(1 / tile.quantizedRange)
-        .add(tile.quantizedVolumeOffset);
-
-      scratchPosition.toArray(decodedArray, i * 3);
-    }
-
-    return decodedArray;
+    scratchPosition
+      .set(positions[i * 3], positions[i * 3 + 1], positions[i * 3 + 2])
+      // TODO - Remove this when deck.gl allows `normalized` flag to be set on `positions`
```

Comment:
What is not working right now? You should be able to pass any attribute-like object as an external buffer.

Reply:
Stray comment. Will remove. This just started working with the recent alpha, which is why I opened the PR.